api/v1/logic.go

```diff
@@ -0,0 +1,121 @@
+package v1
+
+import (
+	"fmt"
+
+	"github.com/oasisprotocol/oasis-core/go/common/cbor"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/accounts"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/consensusaccounts"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/modules/evm"
+	"github.com/oasisprotocol/oasis-sdk/client-sdk/go/types"
+
+	uncagegorized "github.com/oasisprotocol/oasis-indexer/analyzer/uncategorized"
+	"github.com/oasisprotocol/oasis-indexer/storage/client"
+)
+
+func renderRuntimeTransaction(storageTransaction client.RuntimeTransaction) (RuntimeTransaction, error) {
+	var utx types.UnverifiedTransaction
+	if err := cbor.Unmarshal(storageTransaction.Raw, &utx); err != nil {
+		return RuntimeTransaction{}, fmt.Errorf("utx unmarshal: %w", err)
+	}
+	tx, err := uncagegorized.OpenUtxNoVerify(&utx)
+	if err != nil {
+		return RuntimeTransaction{}, fmt.Errorf("utx open no verify: %w", err)
+	}
+	sender0, err := uncagegorized.StringifyAddressSpec(&tx.AuthInfo.SignerInfo[0].AddressSpec)
+	if err != nil {
+		return RuntimeTransaction{}, fmt.Errorf("signer 0: %w", err)
+	}
+	var cr types.CallResult
+	if err = cbor.Unmarshal(storageTransaction.ResultRaw, &cr); err != nil {
+		return RuntimeTransaction{}, fmt.Errorf("result unmarshal: %w", err)
+	}
+	apiTransaction := RuntimeTransaction{
+		Round:   storageTransaction.Round,
+		Index:   storageTransaction.Index,
+		Hash:    storageTransaction.Hash,
+		EthHash: storageTransaction.EthHash,
+		// TODO: Get timestamp from that round's block
+		Sender0:   sender0,
+		Nonce0:    tx.AuthInfo.SignerInfo[0].Nonce,
+		FeeAmount: tx.AuthInfo.Fee.Amount.Amount.String(),
+		FeeGas:    tx.AuthInfo.Fee.Gas,
+		Method:    tx.Call.Method,
+		Body:      tx.Call.Body,
+		Success:   cr.IsSuccess(),
+	}
+	if err = uncagegorized.VisitCall(&tx.Call, &cr, &uncagegorized.CallHandler{
```

Comment:
Is the plan to move this logic to the other files in a follow-up PR?

Reply:
no, do you want to use it elsewhere?

Reply:
wait, referring to the `renderRuntimeTransaction` function or something else?

Reply:
Mainly referring to `renderRuntimeTransaction`; it's pretty gnarly so I initially assumed that the `logic.go` file was a placeholder. After reading through everything though; I understand the structure better now so I'm good with it as-is.